tests/test_surprise.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jun 26 10:42:44 2019
+
+@author: mohamed.cherif
+"""
+
+import logging
+import pickle
+
+import pandas as pd
+import numpy as np
+
+from pytest import mark
+
+try:
+    import surprise
+    have_surprise = True
+except ImportError:
+    have_surprise = False
+
+import lenskit.util.test as lktu
+from lenskit.algorithms.surprise import (SurpriseKNNBasic, SurpriseCoClustering,
+ SurpriseKNNBaseline, SurpriseKNNWithMeans, SurpriseKNNWithZScore, SurpriseNMF,
+ SurpriseNormalPredictor, SurpriseSlopeOne, SurpriseSVD, SurpriseSVDpp)
+
+from lenskit import util
+
+_log = logging.getLogger(__name__)
+
+simple_df = pd.DataFrame({'item': [1, 1, 2, 3],
+                          'user': [10, 12, 10, 13],
+                          'rating': [4.0, 3.0, 5.0, 2.0]})
+
+surprise_algo_list = [SurpriseKNNBasic, SurpriseCoClustering,
+ SurpriseKNNBaseline, SurpriseKNNWithMeans, SurpriseKNNWithZScore, SurpriseNMF,
+ SurpriseNormalPredictor, SurpriseSlopeOne, SurpriseSVD, SurpriseSVDpp]
+# none knn based algos and none normal predictor list for test
+surprise_algo_list_none_knn_based = [SurpriseCoClustering, SurpriseNMF
+                                     , SurpriseSlopeOne, SurpriseSVD, SurpriseSVDpp]
+
+@mark.slow
+@mark.skipif(not have_surprise, reason='surprise not installed')
+def test_surprise_train_rec():
```

Comment:
Since this is operating over each of the algorithms, it looks like a good candidate for `mark.parametrize` to parameterize it over the algorithms. That way each algorithm can fail independently.